cmd/object-handlers.go

```diff
@@ -2711,6 +2711,10 @@ func (api objectAPIHandlers) DeleteObjectHandler(w http.ResponseWriter, r *http.
 
 	opts, err := delOpts(ctx, r, bucket, object)
 	if err != nil {
+		if isErrInvalidVersionID(err) {
+			writeSuccessNoContent(w)
+			return
+		}
```

Comment:
I don't know why would we need this PR @poornas since with AWS S3 

```
~  aws --profile s3 s3api delete-object --bucket harshavardhana --key windows/hosts --version-id XfDr_jqkwLiLdyUJfFRtthkloVRy3oTg 
{
    "VersionId": "XfDr_jqkwLiLdyUJfFRtthkloVRy3oTg"
}

~ aws --profile s3 s3api delete-object --bucket harshavardhana --key windows/hosts --version-id XfDr_jqkwLiLdyUJfFRtthkloVRy3oT1

An error occurred (InvalidArgument) when calling the DeleteObject operation: Invalid version id specified
```

Our behavior here is the same @poornas - not returning an error is perhaps not needed.

Reply:
that was my thought also - @kannappanr was suggesting that delete-object and delete-objects api should be consistent w.r.t minio when invalid version id is specified, as delete-objects does not return invalid version id error. But I see that your PR is probably addressing this better. Will close this.